### PR TITLE
rstudio-server 1.2.5001

### LINKED ISSUE #20: rstudio-server "error 61" after reboot; rebuilding from source works
- [x] Confirmed this is a problem with `brew install`ing one, specific formula and not every time you run `brew`? If it's a general `brew` problem please file this issue at https://github.com/Homebrew/brew/issues/new or https://github.com/Linuxbrew/brew/issues/new.
- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed all issues and retried your prior step?

I'm adding this more as a record in case anyone else hits it, but hopefully it might also be useful to maintainers.

After a recent reboot, rstudio-server stopped working. When visiting it in a browser, after logging in the error message "Unable to connect to service" appears. In the system logs (`/var/log/system.log`) output of the following form appears:

```
Oct 22 13:49:04 iMac rserver[1764]: ERROR system error 61 (Connection refused) [request-uri=/rpc/client_init]; OCCURRED AT: void rstudio::core::http::LocalStreamAsyncClient::handleConnect(const boost::system::error_code &) /tmp/rstudio-server-20190516-24750-15buw5s/rstudio-1.2.1335/src/cpp/core/include/core/http/LocalStreamAsyncClient.hpp:119; LOGGED FROM: void rstudio::server::session_proxy::(anonymous namespace)::logIfNotConnectionTerminated(const rstudio::core::Error &, const http::Request &) /tmp/rstudio-server-20190516-24750-15buw5s/rstudio-1.2.1335/src/cpp/server/ServerSessionProxy.cpp:373
```

`brew doctor` shows no issues, and reinstalling `rstudio-server` from bottle also failed to resolve it. Running `sudo rstudio-server verify-installation` exits with `0`, and `sudo rstudio-server test-config` also exits with `0`. Rebooting multiple times had no effect, and turning off anti-virus also didn't resolve the issue.

The computer is running system version `macOS 10.14.5 (18F132)`.

What ultimately resolved it was rebuilding this formula from source:

```
brew install boost-rstudio-server
brew cask install homebrew/cask-versions/adoptopenjdk8
brew install rstudio-server -s
```

I wonder if anyone else has hit this recently? I also noticed that the version of Xcode being used to build the bottle is `9.0`, whereas the current version of Xcode is at least `11`; might that be related?

Anyway, I hope this is useful to someone.

### PR DESCRIPTION
**Pull request by randy3k**
- [x] Have you followed the [guidelines for contributing](https://github.com/brewsci/homebrew-base/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/brewsci/homebrew-base/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source FORMULA`, where `FORMULA` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict FORMULA` (after doing `brew install FORMULA`)?

-----

It should also fix #20

### CODE REVIEW

**Comment by randy3k** [comment]
The Mac build took too long to finish and the Linux build halted.
I guess we might need to further increase the no output time out limit.

@sjackman do you know what happened to the Linux build?

**Comment by randy3k** [comment]
@sjackman The macOS build is done except that I have to install `adoptopenjdk8` in CircleCI because there is no way to depend on a cask formula. And as usual, the linux build was out of memory and was killed, though I could build it locally.

**Comment by randy3k** [comment]
It looks like the github actions PR is not yet ready to be used. I will take a look in the future when it has been merged.

For this PR, would we just merge it as is? As we don’t know how long would the (GitHub actions be ready).

**Comment by sjackman** [comment]
If you want to merged it without a Linux bottle, then add
```ruby
# Exceeds available memory on CircleCI.
depends_on :macos if ENV["CI"]
```

**Comment by randy3k** [comment]
@sjackman check it again.

File: Formula/rstudio-server.rb [hunk 1]
```diff
@@ -80,15 +84,11 @@ def which_linux_distribution
...
-  # fix build failure with latest macOS compiler
-  patch do
-    url "https://github.com/rstudio/rstudio/commit/b3b1ef10ca8473b6cf02b22cdb57f871f0237a02.diff?full_index=1"
-    sha256 "37c58b7f31307db369ba8c5269f5c303838e06d5db44429766b0213ea6eb6c34"
-  end
-
   def install
-    # Reduce memory usage below 4 GB for CI.
-    ENV["MAKEFLAGS"] = "-j2" if ENV["CIRCLECI"] || ENV["TRAVIS"]
+    if ENV["CIRCLECI"] || ENV["TRAVIS"]
```
Comment (sjackman) [inline Formula/rstudio-server.rb]:
```suggestion
    if ENV["CI"]
```
It looks like this `CIRCLECI` environment variable is being removed by brew. See https://github.com/Homebrew/brew/blob/3944416c4602ae8618c4da399743fe96ddf0cf6a/bin/brew#L82

File: Formula/rstudio-server.rb [hunk 2]
```diff
@@ -11,14 +11,17 @@ class RstudioServer < Formula
...
-    end
     depends_on "patchelf" => :build
     depends_on "libedit"
     depends_on "ncurses"
     depends_on "util-linux" # for libuuid
-    depends_on "linuxbrew/extra/linux-pam"
+    depends_on "linux-pam"
+  end
+
+  if ENV["CIRCLECI"] || ENV["TRAVIS"]
```
Comment (sjackman) [inline Formula/rstudio-server.rb]:
```suggestion
  if ENV["CI"]
```

File: Formula/rstudio-server.rb [hunk 3]
```diff
@@ -11,14 +11,17 @@ class RstudioServer < Formula
...
+    if OS.linux?
+      depends_on "jdk@8" => :build
```
Comment (sjackman) [inline Formula/rstudio-server.rb]:
```suggestion
      depends_on "adoptopenjdk" => :build
```

File: Formula/rstudio-server.rb [hunk 4]
```diff
@@ -30,7 +33,8 @@ class RstudioServer < Formula
   depends_on "cmake" => :build
   depends_on "gcc" => :build
   depends_on :java => ["1.8", :build]
-  depends_on :macos if ENV["CIRCLECI"]
+  # Exceeds available memory on CircleCI.
+  depends_on :macos if ENV["CI"]
```
Comment (sjackman) [inline Formula/rstudio-server.rb]: Want to try removing this line? With the other changes, hopefully it'll build on Linux again.

**Comment by randy3k** [comment]
Wow, it is working now.

**Comment by sjackman** [comment]
For future PRs, the preferred commit log message and pull request title is `rstudio-server 1.2.5001`

**Comment by sjackman** [comment]
Merged. Thanks again for your contribution to Brewsci/base, Randy!

**Pull request closed without merging**